diff --git a/Source/WebCore/platform/audio/DenormalDisabler.cpp b/Source/WebCore/platform/audio/DenormalDisabler.cpp
index 35715e663ce7..44ba08a33d5d 100644
--- a/Source/WebCore/platform/audio/DenormalDisabler.cpp
+++ b/Source/WebCore/platform/audio/DenormalDisabler.cpp
@@ -91,7 +91,7 @@ DenormalDisabler::~DenormalDisabler()
     }
 }
 #else
-DenormalDisabler::DenormalDisabler() = default
+DenormalDisabler::DenormalDisabler() = default;
 DenormalDisabler::~DenormalDisabler() = default;
 #endif
